checker/src/types/regexp.rs

```diff
@@ -0,0 +1,91 @@
+use std::{collections::HashMap, fmt};
+
+use regress::{backends, Flags, Match, Regex};
+
+use crate::BinarySerializable;
+
+#[derive(Debug, Clone)]
+pub struct RegExp {
+	source: String,
+	pub groups: u32,
+	pub named_group_indices: HashMap<String, u16>,
+	re: Regex,
+}
+
+impl RegExp {
+	pub fn new(pattern: &str, flags: Option<&str>) -> Self {
+		let source = if let Some(flags) = flags {
+			format!("/{pattern}/{flags}")
+		} else {
+			format!("/{pattern}/")
+		};
+
+		let mut f = Flags::default();
+
+		if let Some(flags) = flags {
+			for flag in flags.chars() {
+				match flag {
+					'd' => unimplemented!("indices for substring matches are not supported"),
```

Comment:
Is this not something `regress` supports?

Reply:
`regress` does not yet support all flags:
https://github.com/ridiculousfish/regress/blob/9f39399b0e61bee0602c791df8be468e00738100/src/api.rs#L59C1-L89C6